From f65e370e39e22877a119a3b3d801e92ea22ce040 Mon Sep 17 00:00:00 2001
From: "Daniel R. Carvalho" <odanrc@yahoo.com.br>
Date: Sun, 21 Mar 2021 19:31:28 -0300
Subject: [PATCH 225/757] sim,tests: Add a tag for drain-related files

This tag can be used to determine which files are needed
when sim/drain.hh is included in a header file. For
example, when declaring a unit test, this tag makes
the SConscript declaration much simpler.

Change-Id: Ie8a44291a0408090ffbb5b078582d3c5c8d1fd55
Signed-off-by: Daniel R. Carvalho <odanrc@yahoo.com.br>
Reviewed-on: https://gem5-review.googlesource.com/c/public/gem5/+/44108
Reviewed-by: Bobby Bruce <bbruce@ucdavis.edu>
Maintainer: Bobby Bruce <bbruce@ucdavis.edu>
Tested-by: kokoro <noreply+kokoro@google.com>
---
 src/sim/SConscript | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/src/sim/SConscript b/src/sim/SConscript
index c0951f3..bf46ccb 100644
--- a/src/sim/SConscript
+++ b/src/sim/SConscript
@@ -52,10 +52,11 @@ Source('cxx_config.cc')
 Source('cxx_manager.cc')
 Source('cxx_config_ini.cc')
 Source('debug.cc')
+Source('drain.cc', add_tags='gem5 drain')
 Source('py_interact.cc', add_tags='python')
 Source('eventq.cc', add_tags='gem5 events')
 Source('futex_map.cc')
-Source('global_event.cc')
+Source('global_event.cc', add_tags='gem5 drain')
 Source('globals.cc')
 Source('init.cc', add_tags='python')
 Source('init_signals.cc')
@@ -66,9 +67,8 @@ Source('python.cc', add_tags='python')
 Source('redirect_path.cc')
 Source('root.cc')
 Source('serialize.cc', add_tags='gem5 serialize')
-Source('drain.cc')
 Source('se_workload.cc')
-Source('sim_events.cc')
+Source('sim_events.cc', add_tags='gem5 drain')
 Source('sim_object.cc')
 Source('sub_system.cc')
 Source('ticked_object.cc')
@@ -89,6 +89,7 @@ Source('stats.cc')
 Source('workload.cc')
 Source('mem_pool.cc')
 
+env.TagImplies('gem5 drain', ['gem5 events', 'gem5 trace'])
 env.TagImplies('gem5 events', ['gem5 serialize', 'gem5 trace'])
 env.TagImplies('gem5 serialize', 'gem5 trace')
 
-- 
1.8.3.1

